### PR TITLE
PR 1264 utils.js feedback refactor

### PR DESCRIPTION
**Pull request by sabyamon**
Resolves utils.js feedback in PR https://github.com/adobecom/milo/pull/1264

Test URLs:

Before: https://uar-integration--milo--adobecom.hlx.page/drafts/xiasun/quiz-2/?martech=off
After: https://uar-utils-feedback--milo--sabyamon.hlx.page/drafts/xiasun/quiz-2/?martech=off

### CODE REVIEW

File: libs/blocks/quiz/utils.js [hunk 1]
```diff
@@ -293,14 +290,13 @@ export const parseResultData = async (answers) => {
...
   ? selectedDestination : defaultValue);
 
-// TODO: needs refactoring - can split to smaller functions
 export const findMatchForSelections = (results, selections) => {
   const recommendations = [];
   const matchResults = [];
   const defaultResult = [];
 
   results.forEach((destination) => {
-    if (destination.result.indexOf('(') === -1) {
```
Comment (JackySun9) [inline libs/blocks/quiz/utils.js:303]: I am not sure for this change, I need to run some tests

**Review by fullcolorcoder — APPROVED** [review]
Looks good to me!

File: libs/blocks/quiz/utils.js [hunk 2]
```diff
@@ -155,10 +155,9 @@ export const structuredFragments = (
         if (umbrellaProduct && row.product === umbrellaProduct) {
           structureFragments.push(row[fragment]);
         }
-      } else if ((primaryProducts.length > 0 && primaryProducts.includes(row.product))) {
-        if (row[fragment]) {
-          structureFragments.push(row[fragment]);
-        }
+      } else if ((primaryProducts.length > 0 && primaryProducts.includes(row.product))
+      && row[fragment]) {
```
Comment (chrischrischris) [inline libs/blocks/quiz/utils.js:159]: nit: indent

**Review by chrischrischris — APPROVED** [review]

**Review by JackySun9 — APPROVED** [review]
test it, looks good

File: libs/blocks/quiz/utils.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sabyamon) [reply libs/blocks/quiz/utils.js]: Updated

**Pull request merged**